askomics/libaskomics/rdfdb/ResultsBuilder.py

```diff
@@ -55,11 +55,13 @@ def organize_attribute_and_entity(self, results, constraints): # FIXME results i
                 entity_attribute_list[entity_var][attr_var] = attribute_name
 
         #removing all empty node. means they are category...
-        for elem in entity_attribute_list.values():
-            for elt_att in elem:
-                if elt_att in  entity_attribute_list:
-                    del entity_attribute_list[elt_att]
-                    del entity_name_list[elt_att]
+        remove_item = [ elt_att
```

Comment:
No, I came up with this while trying to understood the logic, but I couldn't find a better way to express this.  My previous code was changing the dict shape while iterating on it, causing a error.
I can revert to original if functional style is a problem.
Ideally categories should be treated separately...